### ISSUE TITLE
Add controlled type generation

### ISSUE THREAD

**Issue by rlouf** [outlines-dev/outlines#90]
We should be able to limit the output of the model to a given type (int, float, str, bool) by applying a mask to the logits before generation:

```python
import outlines.models as models

complete = models.text_completion.hf("sshleifer/tiny-gpt2")

prompt = "Give me a number between 0 and 10"
answer = complete(prompt, type=int)
```
